keel-core/src/main/kotlin/com/netflix/spinnaker/keel/constraints/ManualJudgementConstraintEvaluator.kt

```diff
@@ -0,0 +1,62 @@
+package com.netflix.spinnaker.keel.constraints
+
+import com.netflix.spinnaker.keel.api.ConstraintState
+import com.netflix.spinnaker.keel.api.ConstraintStatus
+import com.netflix.spinnaker.keel.api.DeliveryArtifact
+import com.netflix.spinnaker.keel.api.DeliveryConfig
+import com.netflix.spinnaker.keel.api.Environment
+import com.netflix.spinnaker.keel.api.ManualJudgementConstraint
+import com.netflix.spinnaker.keel.constraints.ConstraintEvaluator.Companion.getConstraintForEnvironment
+import com.netflix.spinnaker.keel.persistence.DeliveryConfigRepository
+import org.springframework.stereotype.Component
+import java.time.Clock
+
+@Component
+class ManualJudgementConstraintEvaluator(
+  private val deliveryConfigRepository: DeliveryConfigRepository,
+  private val clock: Clock
+) : StatefulConstraintEvaluator<ManualJudgementConstraint> {
+
+  override val constraintType = ManualJudgementConstraint::class.java
+
+  override fun canPromote(
+    artifact: DeliveryArtifact,
+    version: String,
+    deliveryConfig: DeliveryConfig,
+    targetEnvironment: Environment,
+    priorState: ConstraintState?
+  ): Boolean {
+    val constraint = getConstraintForEnvironment(deliveryConfig, targetEnvironment.name, constraintType)
+
+    val state = priorState ?: ConstraintState(
```

Comment:
I presume we’ll eventually want to send a notification if `priorState == null` so that users know to do something?

Reply:
That makes sense to me as well.

Reply:
The `.also` block this comment is above already contains:
```
          // TODO: Emit an event here, requesting a manual judgement
```
which I think is clear.